azure-iot-hub/azure/iot/hub/iothub_amqp_client.py

```diff
@@ -70,20 +113,24 @@ def send_message_to_device(self, device_id, message, app_props):
         message = uamqp.Message(
             msg_content, properties=msg_props, application_properties=app_properties
         )
-        self.amqp_client.queue_message(message)
-        results = self.amqp_client.send_all_messages(close_on_done=False)
+        self.amqp_message_send_client.queue_message(message)
+        results = self.amqp_message_send_client.send_all_messages(close_on_done=False)
         if uamqp.constants.MessageState.SendFailed in results:
             raise Exception("C2D message send failure")
```

Comment:
One thing that @cartertinney pointed out is that it's a bad idea to raise an `Exception` here and we should probably raise something different. This PR might be a good place to do it, though I'm not sure what the best thing to raise would be.

Reply:
Yep, that's one of about 6 reasons that I"m pushing to deprecate this function.